zerver/lib/actions.py

```diff
@@ -2636,13 +2639,38 @@ def do_change_bot_owner(user_profile: UserProfile, bot_owner: UserProfile,
     RealmAuditLog.objects.create(realm=user_profile.realm, acting_user=acting_user,
                                  modified_user=user_profile, event_type='bot_owner_changed',
                                  event_time=event_time)
+
+    update_users = bot_owner_user_ids(user_profile)
+
+    # For admins, update event is sent instead of delete/add
+    # event. bot_data of admin contains all the
+    # bots and none of them should be removed/(added again).
+
+    # Notify the new owner that the bot has been added.
+    if not bot_owner.is_realm_admin:
+        add_event = created_bot_event(user_profile)
+        send_event(add_event, {bot_owner.id, })
+        # Do not send update event for bot_owner.
+        update_users = update_users - {bot_owner.id, }
+
+    # Delete the bot from previous owner's bot data.
+    if not acting_user.is_realm_admin:
```

Comment:
`acting_user` may not be the previous owner -- e.g. an admin can re-assign the owner.

I think you want to save  `previous_owner` variable above (before we overwrite it) and use that here.